_source/_posts/2023-07-27-spring-graphql-react.adoc

```diff
@@ -0,0 +1,1857 @@
+
+**NOTE**: My browser was not displaying anything, so I had to manually activate the device by opening the URL `https://auth0.auth0.com/activate?user_code={deviceCode}`.
+
+On successful login, you will see the tenant, which you will use as the issuer later:
+
+----
+✪ Welcome to the Auth0 CLI 🎊
+
+If you don't have an account, please create one here: https://auth0.com/signup.
+
+Your device confirmation code is: KGFL-LNVB
+
+ ▸    Press Enter to open the browser to log in or ^C to quit...
+
+Waiting for the login to complete in the browser... ⣻Opening in existing browser session.
+Waiting for the login to complete in the browser... done
+
+ ▸    Successfully logged in.
+ ▸    Tenant: dev-avup2laz.us.auth0.com
+----
+
+The next step is to create a client app, which you can do in one command:
+
+[source, shell]
+----
+auth0 apps create \
+  --name "GraphQL Server" \
+  --description "Spring Boot GraphQL Resource Server" \
+  --type regular \
+  --callbacks http://localhost:8080/login/oauth2/code/okta \
+  --logout-urls http://localhost:8080 \
+  --reveal-secrets
+----
+
+Once the app is created, you will see the OIDC app's configuration:
+
+----
+=== dev-avup2laz.us.auth0.com application created
+
+  CLIENT ID            ***
+  NAME                 GraphQL Server
+  DESCRIPTION          Spring Boot GraphQL Resource Server
+  TYPE                 Regular Web Application
+  CLIENT SECRET        ***
+  CALLBACKS            http://localhost:8080/login/oauth2/code/okta
+  ALLOWED LOGOUT URLS  http://localhost:8080
+  ALLOWED ORIGINS
+  ALLOWED WEB ORIGINS
+  TOKEN ENDPOINT AUTH
+  GRANTS               implicit, authorization_code, refresh_token, client_credentials
+
+ ▸    Quickstarts: https://auth0.com/docs/quickstart/webapp
+ ▸    Hint: Emulate this app's login flow by running `auth0 test login ***`
+ ▸    Hint: Consider running `auth0 quickstarts download ***`
+----
+
+Add the `okta-spring-boot-starter` dependency to the `build.gradle` file:
+----
+dependencies {
+    ...
+  implementation 'com.okta.spring:okta-spring-boot-starter:3.0.4'
+    ...
+}
+----
+
+Set the clientId, issuer and audience for OAuth2 in the `application.properties` file:
+
+.application.properties
+[source, properties]
+----
+okta.oauth2.issuer=https://{yourAuth0Domain}/
+okta.oauth2.client-id={clientId}
+okta.oauth2.audience=https://{yourAuth0Domain}/api/v2/
+----
+
+Add the clientSecret to the `.env` file:
+
+..env
+[source, shell]
+----
+export OKTA_OAUTH2_CLIENT_SECRET={clientSecret}
+----
+
+Add the following factory method to the class `SpringBootGraphQLApiConfig`, for requiring a bearer token for all requests:
+
+.SpringBootGraphQLApiConfig.java
+[source, java]
+----
+    ...
+    @Bean
+    public SecurityFilterChain configure(HttpSecurity http) throws Exception {
+        http.oauth2ResourceServer(oauth2ResourceServer -> oauth2ResourceServer.jwt(withDefaults()));
+        return http.build();
+    }
+    ...
+----
```

Comment:
I believe this is already enabled by the Okta Spring Boot starter. I don't think it's necessary.

https://github.com/okta/okta-spring-boot/blob/master/oauth2/src/main/java/com/okta/spring/boot/oauth/OktaOAuth2AutoConfig.java#L88

Reply:
If I remove it, it seems the cors configuration at GraphQLConfig does not make effect, and requests from the react client fail. Not sure why is this happening.

Reply:
If that's the case, maybe add a note about why it's necessary.

Reply:
Tested again, and added the note.